src/eXpansion/Framework/Notifications/Plugins/Gui/NotificationWidget.php

```diff
@@ -0,0 +1,201 @@
+	
+	
+	foreach (Read in Page.GetClassChildren_Result) {
+		
+		declare Frame = (Read as CMlFrame);
+		
+		// stop at first available toast
+		if ( (Frame.Controls[0] as CMlLabel).Opacity == 0.) {		
+		Frame.Show();
+		Frames.add(Frame);
+		(Frame.Controls[0] as CMlLabel).Opacity = 0.1;	
+		(Frame.Controls[1] as CMlLabel).Value = Title;
+		(Frame.Controls[2] as CMlLabel).Value = Message;
+		
+			foreach(Element in Frame.Controls) {
+					AnimMgr.Add(Element, "<elem opacity=\"1\" />", 500, CAnimManager::EAnimManagerEasing::QuadIn);
+					if (Duration > 0) {
+						AnimMgr.Add(Element, "<elem opacity=\"0\" />", (Now + Duration), 500, CAnimManager::EAnimManagerEasing::QuadOut);
+					}
+			}
+						
+			return;
+		}
+	}
+}
+
+EOL
+        );
+
+        $manialink->getFmlManialink()->getScript()->addCustomScriptLabel(ScriptLabel::OnInit, <<<EOL
+                
+                
+                    declare CMlFrame[] Frames for This;
+                    Frames.clear();
+                    Page.GetClassChildren("uiToast", (Page.GetFirstChild("Window") as CMlFrame), True);
+                  
+                    foreach (Frame in Page.GetClassChildren_Result) {
+
+                        foreach(Element in (Frame as CMlFrame).Controls) {
+                            if (Element is CMlLabel) {
+                                (Element as CMlLabel).Opacity = 0.;
+                            }
+                            
+                            if (Element is CMlQuad) {
+                                (Element as CMlQuad).Opacity = 0.;
+                            }
+                        }
+                    }
+                    {$this->updaterWidgetFactory->getScriptInitialization(true)}
+EOL
+        );
+        $manialink->getFmlManialink()->getScript()->addCustomScriptLabel(ScriptLabel::Loop, <<<EOL
+
+             {$this->updaterWidgetFactory->getScriptOnChange('
+             if (notification.count > 0) {
+                 ShowToast(notification["title"], notification["message"], TextLib::ToInteger(notification["duration"]));
+                 }
+             ')}
+             
+                declare Index = 0;
+                foreach(Frame in Frames) {
+                    declare Real posY = 70.;		
+                    if ((Frame.Controls[0] as CMlLabel).Opacity > 0.09) {
+                        Frame.RelativePosition_V3.Y = posY - (Index* 28. * 0.8);
+                        Index +=1;
+                    } else {
+                        HideToast(Frame, True);
+                        Frame.Hide();
+                    }
+                }
+                  
+EOL
+        );
+
+        $manialink->getFmlManialink()->getScript()->addCustomScriptLabel(ScriptLabel::MouseClick, <<<EOL
+         if (Event.Control.HasClass("toastClose")) {
+			HideToast(Event.Control.Parent, False);
+		}
+        
+EOL
+        );
+
+
+    }
+
+
+    private function createToast($idx)
+    {
```

Comment:
So toast are not translatable ?

Reply:
Oh, forgot to add ->setTranslate(true)....

Reply:
Dang. Notifications aren't translatable. I don't see how I can pass multilingual data to the toast viewer from the updater.

Reply:
I managed to get the translations working using maniascript magic.